service/history/historyEngine.go

```diff
@@ -268,20 +268,12 @@ func (e *historyEngineImpl) Start() {
 	e.logger.Info("", tag.LifeCycleStarting)
 	defer e.logger.Info("", tag.LifeCycleStarted)
 
+	e.registerNamespaceStateChangeCallback()
+
 	for _, queueProcessor := range e.queueProcessors {
 		queueProcessor.Start()
 	}
 	e.replicationProcessorMgr.Start()
-
-	// failover callback will try to create a failover queue processor to scan all inflight tasks
-	// if domain needs to be failovered. However, in the multicursor queue logic, the scan range
-	// can't be retrieved before the processor is started. If failover callback is registered
-	// before queue processor is started, it may result in a deadline as to create the failover queue,
-	// queue processor need to be started.
-	//
-	// Ideally, when both timer and transfer queues enabled single cursor mode, we don't have to register
-	// the callback. However, currently namespace migration is relying on the callback to UpdateHandoverNamespaces
-	e.registerNamespaceFailoverCallback()
```

Comment:
This change is from Cadence (https://github.com/uber/cadence/pull/3494) and should not be [picked](https://github.com/temporalio/temporal/pull/754). It only applies to cadence version of multi-cursor implementation which still has the concept for failover queue.